### PR TITLE
Fixed the useg_subnet logic in the mso_schema_template_anp_epg_useg_attribute.py module

### PR DESCRIPTION
**Pull request by sajagana**
Note:

Fixed the useg_subnet logic and added additional validation tasks

### CODE REVIEW

File: plugins/modules/mso_schema_template_anp_epg_useg_attribute.py [hunk 1]
```diff
@@ -229,11 +229,13 @@ def main():
         payload = dict(name=name, displayName=name, description=description, type=EPG_U_SEG_ATTR_TYPE_MAP[attribute_type], value=value)
 
         if attribute_type == "ip":
-            if useg_subnet is False:
+            if useg_subnet:
+                if value != "" and value != "0.0.0.0" and value is not None:
+                    mso.fail_json(msg="The value should be empty string or 0.0.0.0, when the useg_subnet is true.")
```
Comment (shrsr) [inline plugins/modules/mso_schema_template_anp_epg_useg_attribute.py:234]: mso.fail_json(msg="The value of uSeg should be an empty string or 0.0.0.0, when the useg_subnet is set to true.")

File: tests/integration/targets/mso_schema_template_anp_epg_useg_attribute/tasks/main.yml [hunk 2]
```diff
@@ -90,6 +90,24 @@
...
+  ignore_errors: true
+  register: ans_test_uSeg_1_nt1
+
+- name: Assertions check for invalid tests - ans_test_uSeg_1
+  ansible.builtin.assert:
+    that:
+      - ans_test_uSeg_1_nt is not changed
+      - ans_test_uSeg_1_nt.msg == "state is present but all of the following are missing{{':'}} name, type"
+      - ans_test_uSeg_1_nt1 is not changed
+      - ans_test_uSeg_1_nt1.msg == "The value should be empty string or 0.0.0.0, when the useg_subnet is true."
```
Comment (shrsr) [inline tests/integration/targets/mso_schema_template_anp_epg_useg_attribute/tasks/main.yml:109]: - ans_test_uSeg_1_nt1.msg == "The value of uSeg should be an empty string or 0.0.0.0, when the useg_subnet is set to true."

**Review by shrsr — CHANGES_REQUESTED** [review]

**Review by samiib — DISMISSED** [review]
Just considering @shrsr's suggestion makes sense, everything else LGTM. 😄

**Review by akinross — APPROVED** [review]
LGTM

**Review by lhercot — APPROVED** [review]
LGTM

**Pull request merged**